crates/turborepo-lib/src/global_deps_package_detector.rs

```diff
@@ -0,0 +1,146 @@
+            // the path is actually in globalDeps and if not, return it as
+            // PackageDetection::Package(WorkspacePackage::root()).
+            PackageDetection::All => {
+                let cleaned_path = path.clean();
+                let in_global_deps = self.global_deps_matcher.is_match(cleaned_path.as_str());
+
+                if in_global_deps {
+                    PackageDetection::All
+                } else {
+                    PackageDetection::Package(WorkspacePackage::root())
+                }
+            }
+            result => result,
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use tempfile::tempdir;
+    use turbopath::{AbsoluteSystemPath, AnchoredSystemPathBuf};
+    use turborepo_repository::{
+        change_mapper::{ChangeMapper, DefaultPackageDetector, PackageChanges},
+        discovery,
+        discovery::PackageDiscovery,
+        package_graph::{PackageGraphBuilder, WorkspacePackage},
+        package_json::PackageJson,
+    };
+
+    use super::GlobalDepsPackageDetector;
+    use crate::turbo_json::TurboJson;
+
+    #[allow(dead_code)]
+    pub struct MockDiscovery;
+
+    impl PackageDiscovery for MockDiscovery {
+        async fn discover_packages(
+            &self,
+        ) -> Result<discovery::DiscoveryResponse, discovery::Error> {
+            Ok(discovery::DiscoveryResponse {
+                package_manager: turborepo_repository::package_manager::PackageManager::Npm,
+                workspaces: vec![],
+            })
+        }
+
+        async fn discover_packages_blocking(
+            &self,
+        ) -> Result<discovery::DiscoveryResponse, discovery::Error> {
+            self.discover_packages().await
+        }
+    }
+
+    #[tokio::test]
+    async fn test_different_package_detectors() -> Result<(), anyhow::Error> {
+        let repo_root = tempdir()?;
+        let root_package_json = PackageJson::default();
+
+        let pkg_graph = PackageGraphBuilder::new(
+            AbsoluteSystemPath::from_std_path(repo_root.path())?,
+            root_package_json,
+        )
+        .with_package_discovery(MockDiscovery)
+        .build()
+        .await?;
+
+        let default_package_detector = DefaultPackageDetector::new(&pkg_graph);
+        let change_mapper = ChangeMapper::new(&pkg_graph, vec![], default_package_detector);
+
+        let package_changes = change_mapper.changed_packages(
+            [AnchoredSystemPathBuf::from_raw("README.md")?]
+                .into_iter()
+                .collect(),
+            None,
+        )?;
+
+        // We should return All because we don't have global deps and
+        // therefore must be conservative about changes
+        assert_eq!(package_changes, PackageChanges::All);
+
+        let turbo_json = TurboJson::default();
+        let turbo_package_detector = GlobalDepsPackageDetector::new(
+            &pkg_graph,
+            turbo_json.global_deps.iter().map(|s| s.as_str()),
```

Comment:
Nit: can we just the value here instead of depending on the default value of `TurboJson`?